### DISCUSSION TITLE
i18n management

### DISCUSSION THREAD

**Discussion by xibman** [sofn-xyz/mailing#244] [Ideas]
Hi another thing that can be really great is to have i18n management.

it's already possible to do something with react i18next with the usage of [i18nextProvider](https://react.i18next.com/latest/i18nextprovider).

But it be great to have this with a better implementation

**Comment by psugihara** [comment]
That's a cool idea @xibman. Any ideas for what you'd like to see in an implementation of i18n in Mailing or how you'd like to use it?

I haven't done much email internationalization so I think I'd need some design help for this feature.

**Reply by xibman** [reply]
Hi @psugihara 

I already have an implementation of full i18n email translation based on your work and [React i18next](https://react.i18next.com/).

It's really simple and it work well. I can share it with you.

It think if you don't want to include a dedicated lib for i18n management you can just provide a way to push the language you want to show to an email.

So on the emailing project side you just have three things:

- an array of language provided by the user that follow this format or an other `['en-GB','es-ES'.......]` and the default one 
- a dropdown menu that operate as a language switcher
- the current language information pass to the email template

So every one could had the lib he want for i18n management

**Reply by psugihara** [reply]
Got it. Yeah I like the idea of having an array of languages in your mailing.config.json that we could consume in the CLI to make the dropdown. We'd only show the dropdown if the language array were provided.

>  I can share it with you.

Would love to check it out!

**Reply by xibman** [reply]
To simplify ui stuff development you can use https://headlessui.com/ .
I see the project use tailwind behind the scene so it can simplify the development of a dropdown or stuff like this.
For the flag icons : https://flagicons.lipis.dev/
For language validation : https://www.npmjs.com/package/countries-list

**Reply by xibman** [reply]
@psugihara i promise i send a repository tomorow with all the stuff inside 

Thanks again for your work !

**Reply by xibman** [reply]
@psugihara  i just share withou a repository why the i18n implementation.

**Reply by psugihara** [reply]
very cool. thank you!
